src/main/webapp/board_state_visualization.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Write hard-coded board states for db.
+ */
+function writeHardCodedBoardStates() {
+    fetch('/write-hard-coded-board-state');
+}
+
+
+/**
+ * Display 10 rounds of a dummy hard-coded simultaion, both real and estimated board.
+ */
+async function fetchHardCodedBoardStates() {
+
+    var simulationId = "demo-tracing-sim-1";
+
+    for (var roundNumber = 0; roundNumber < 10; roundNumber++) {
+        // Set params to POST request.
+        const params = new URLSearchParams();
+        params.append('simulationId', 'demo-tracing-sim-1');
+        params.append('simulationId', simulationId);
+        params.append('round', roundNumber);
+
+        // Fetch and display real board state.
+        fetch('/read-hard-coded-real-board-state', {method: 'POST', body: params})
+        .then(response => response.json())
+        .then(boardState => visualizeHardCodedBoardState(boardState.array, true));
+
+        // Fetch and display real board state.
+        fetch('/read-hard-coded-estimated-board-state', {method: 'POST', body: params})
+        .then(response => response.json())
+        .then(boardState => visualizeHardCodedBoardState(boardState.array, false));
+
+        await sleep(1000);
+    }
+}
+
+
+/**
+ * JS sleep method implementation.
+ * @param {number} ms number of milliseconds to sleep.
+ */
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+
+/**
+ * Visualize the board state on index.html page.
+ * @param {boardState} board a JSON string representing the board state in a certain round of the simulation.
+ * @param {boolean} isReal   if 'true' visualize the real board, otherwise visualize the estimated board.
+ */
+function visualizeHardCodedBoardState(board, isReal) {
```

Comment:
I'll quote Alon:
"Google best practices recommend to avoid boolean parameters as much as possible; instead, we recommend to use two separate methods, whose names reflect the effect of the boolean, and possibly sharing aspects of the implementation. (The reasons to avoid boolean parameters are (a) they're hard to extend, and (b) they more-or-less guarantee higher complexity - more "if"s - within the method)."